[Rock] Code Spring Clean Up

Soare-Robert-Daniel opened the pull request:
<!-- Issues that this pull request closes. -->
Closes #1635.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

### Summary
<!-- Please describe the changes you made. -->

Clean the code:
- Remove ad-hoc functions
- Make some functions reusable
- Simplify small parts

### Screenshots <!-- if applicable -->

----

### Test instructions
<!-- Describe how this pull request can be tested. -->

<!--
#### Query
```javascript
new QueryQA().select('blocks').run()
```
-->

---- 

### Checklist before the final review

- [ ] Visual elements are not affected by independent changes.
- [ ] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [ ] It loads additional script in frontend only if it is required.
- [ ] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [ ] In case of deprecation, old blocks are safely migrated.
- [ ] It is usable in Widgets and FSE.
- [ ] Copy/Paste is working if the attributes are modified.

irinelenache commented:
@Soare-Robert-Daniel Tested the PR and here is what i found:

- [ ] Switching to the style tab for the **Flip** block is breaking the block and throwing a console error https://vertis.d.pr/v/4BLogo
- [ ] ^ Same happens for the **Posts** block
- [ ] I positioned the **Slider** block after the **product review** and the **Sync with woocommerce** setting is visible in the Slider block https://vertis.d.pr/i/kAxVBf (I saw it in other blocks too)
- [ ] This is how it looks when i'm editing the Popup and there is an accordion after https://vertis.d.pr/i/Kgy6Pt

Soare-Robert-Daniel commented:
@irinelenache, those issue were introduced by the changes and are now fixed 👌 
> * [x]  Switching to the style tab for the **Flip** block is breaking the block and throwing a console error https://vertis.d.pr/v/4BLogo
> * [x]  ^ Same happens for the **Posts** block

Those should not happen with the changes made.

> * [ ]  I positioned the **Slider** block after the **product review** and the **Sync with woocommerce** setting is visible in the Slider block https://vertis.d.pr/i/kAxVBf (I saw it in other blocks too)
> * [ ]  This is how it looks when i'm editing the Popup and there is an accordion after https://vertis.d.pr/i/Kgy6Pt

I could not reproduce them

https://github.com/Codeinwp/otter-blocks/assets/17597852/78a60fff-c084-4341-a5c2-034fb4527d42

Please send the instance if you can reproduce them with the new build 🙏

irinelenache commented:
@Soare-Robert-Daniel The marked issues are fixed, thank you 🚀 

Here is the instance for the other 2:
```
 Admin area URL: https://overjoyedveil.s4-tastewp.com/2023/05/11/previously-created-blocks/
 Username: irinel 
 Password: uyWGqFCffxc 
```
Let me know if the issues happen for you too or there's an issue on my end only 🙏

Soare-Robert-Daniel commented:
@irinelenache, I got the reproduce them 👌  and fixed 🔨 

@HardeepAsrani, the issue highlighted by Irinel where pretty trick.

The first one regarding the `Sync with WooComerce` happened because the hook was attacking the component using the `Slot` and not `InspectorControls`. By using `Slot`, that component keeps hanging out to other blocks.

Commit: https://github.com/Codeinwp/otter-blocks/pull/1649/commits/f28a5148124e6bf05a9ab76ea363b594f6af8b19

For targeted hooks, we should not use `window.otterComponents.useInspectorSlot`

The second, this one was very tricky since when you pop up a model, you need to make sure that the parent has also a `z-index`. Without taking the consideration the context, the accordion with `z-index: 2` was on top of the Modal with `z-index: 99999`. Added this link to our internal docs for future reference: https://www.freecodecamp.org/news/4-reasons-your-z-index-isnt-working-and-how-to-fix-it-coder-coder-6bc05f103e6c/

Commit: https://github.com/Codeinwp/otter-blocks/pull/1649/commits/23031cc14661b905ca466364fb73adfe07ea7d5d

irinelenache commented:
@Soare-Robert-Daniel All issues are fixed now, thank you 🚀
